blockcache/blockcache_test.go

```diff
@@ -20,35 +20,47 @@ type mockChainBackend struct {
 	sync.RWMutex
 }
 
-func (m *mockChainBackend) addBlock(block *wire.MsgBlock, nonce uint32) {
-	m.Lock()
-	defer m.Unlock()
-	block.Header.Nonce = nonce
-	hash := block.Header.BlockHash()
-	m.blocks[hash] = block
+func newMockChain() *mockChainBackend {
+	return &mockChainBackend{
+		blocks: make(map[chainhash.Hash]*wire.MsgBlock),
+	}
 }
+
+// GetBlock is a mock implementation of block fetching that tracks the number
+// of backend calls and returns the block found for the given hash or an error.
 func (m *mockChainBackend) GetBlock(blockHash *chainhash.Hash) (*wire.MsgBlock, error) {
-	m.RLock()
-	defer m.RUnlock()
+	m.Lock()
+	defer m.Unlock()
+
 	m.chainCallCount++
 
 	block, ok := m.blocks[*blockHash]
+
 	if !ok {
 		return nil, fmt.Errorf("block not found")
 	}
 
 	return block, nil
 }
 
-func newMockChain() *mockChainBackend {
-	return &mockChainBackend{
-		blocks: make(map[chainhash.Hash]*wire.MsgBlock),
-	}
+func (m *mockChainBackend) getChainCallCount() int {
+	m.RLock()
+	defer m.RUnlock()
+
+	return m.chainCallCount
+}
+
+func (m *mockChainBackend) addBlock(block *wire.MsgBlock, nonce uint32) {
+	m.Lock()
+	defer m.Unlock()
```

Comment:
since we are moving this around, pls can you add a new line after the defer statement?